### PR TITLE
[CMake] Clean modules post-NG

### PR DESCRIPTION
**Pull request by guparan**
More post-NG cleaning.
Follows #2958

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: SofaKernel/modules/Sofa.GL/Component/Engine/src/sofa/gl/component/engine/config.h.in [hunk 1]
```diff
@@ -34,8 +34,8 @@
 // DEPRECATION MACROS
 // Keep the previous macros
 // This backward compatibility will be removed at v23.06
-#ifndef SOFAGENERALENGINE_HAVE_SOFA_GL
-#define SOFAGENERALENGINE_HAVE_SOFA_GL 1
+#ifndef SOFAGENERALENGINE_HAVE_SOFA_GL_COMPONENT_ENGINE
```
Comment (fredroy) [inline SofaKernel/modules/Sofa.GL/Component/Engine/src/sofa/gl/component/engine/config.h.in:40]: Why the change ? These definitions should not be modified I think. (as they are used for compatibility)

File: modules/SofaGeneralEngine/src/SofaGeneralEngine/config.h.in [hunk 2]
```diff
@@ -24,7 +24,7 @@
 #include <sofa/config.h>
 #include <sofa/config/sharedlibrary_defines.h>
 
-#define SOFAGENERALENGINE_HAVE_SOFA_GL @SOFAGENERALENGINE_HAVE_SOFA_GL@
+#define SOFAGENERALENGINE_HAVE_SOFA_GL_COMPONENT_ENGINE @SOFAGENERALENGINE_HAVE_SOFA_GL_COMPONENT_ENGINE@
```
Comment (fredroy) [inline modules/SofaGeneralEngine/src/SofaGeneralEngine/config.h.in:30]: same here, people were (maybe) using SOFAGENERALENGINE_HAVE_SOFA_GL variable. But if it is is renamed, this is new; so the user will have an error.

**Review by fredroy — CHANGES_REQUESTED** [review]

File: SofaKernel/modules/Sofa.GL/Component/Engine/src/sofa/gl/component/engine/config.h.in (hunk 1 above)
Comment (guparan) [reply SofaKernel/modules/Sofa.GL/Component/Engine/src/sofa/gl/component/engine/config.h.in:40]: Wanted to be consistent with what is actually needed. I will re-add the compat :+1:

**Review by fredroy — APPROVED** [review]
